src/chains/ethereum/block/src/block.ts

```diff
@@ -156,44 +121,68 @@ export class Block {
-    if (!common.isActivatedEIP(1559)) {
-      return undefined;
-    }
+  static calcNextBaseFeeBigInt(parentHeader: BaseFeeHeader) {
     let nextBaseFee: bigint;
-    // genesis block
-    if (parentBlock === undefined) {
-      nextBaseFee = BlockParams.INITIAL_BASE_FEE_PER_GAS;
-    } else {
-      const header = parentBlock.header;
-      const parentGasTarget =
-        header.gasLimit.toBigInt() / BlockParams.ELASTICITY;
-      const parentGasUsed = header.gasUsed.toBigInt();
-      const baseFeePerGas = header.baseFeePerGas
-        ? header.baseFeePerGas.toBigInt()
-        : BlockParams.INITIAL_BASE_FEE_PER_GAS;
-      if (parentGasTarget === parentGasUsed) {
-        nextBaseFee = baseFeePerGas;
-      } else if (parentGasUsed > parentGasTarget) {
-        const gasUsedDelta = parentGasUsed - parentGasTarget;
-        const adjustedFeeDelta =
-          (baseFeePerGas * gasUsedDelta) /
-          parentGasTarget /
-          BlockParams.BASE_FEE_MAX_CHANGE_DENOMINATOR;
-        if (adjustedFeeDelta > 1n) {
-          nextBaseFee = baseFeePerGas + adjustedFeeDelta;
-        } else {
-          nextBaseFee = baseFeePerGas + 1n;
-        }
+
+    const header = parentHeader;
+    const parentGasTarget = header.gasLimit.toBigInt() / BlockParams.ELASTICITY;
+    const parentGasUsed = header.gasUsed.toBigInt();
+    const baseFeePerGas = header.baseFeePerGas
+      ? header.baseFeePerGas.toBigInt()
+      : BlockParams.INITIAL_BASE_FEE_PER_GAS;
+
+    if (parentGasTarget === parentGasUsed) {
+      // If the parent gasUsed is the same as the target, the baseFee remains unchanged.
+      nextBaseFee = baseFeePerGas;
+    } else if (parentGasUsed > parentGasTarget) {
+      // If the parent block used more gas than its target, the baseFee should increase.
+      const gasUsedDelta = parentGasUsed - parentGasTarget;
+      const adjustedFeeDelta =
+        (baseFeePerGas * gasUsedDelta) /
+        parentGasTarget /
+        BlockParams.BASE_FEE_MAX_CHANGE_DENOMINATOR;
+      if (adjustedFeeDelta > 1n) {
+        nextBaseFee = baseFeePerGas + adjustedFeeDelta;
       } else {
-        const gasUsedDelta = parentGasTarget - parentGasUsed;
-        const adjustedFeeDelta =
-          (baseFeePerGas * gasUsedDelta) /
-          parentGasTarget /
-          BlockParams.BASE_FEE_MAX_CHANGE_DENOMINATOR;
-        nextBaseFee = baseFeePerGas - adjustedFeeDelta;
+        nextBaseFee = baseFeePerGas + 1n;
       }
+    } else {
+      // Otherwise if the parent block used less gas than its target, the baseFee should decrease.
+      const gasUsedDelta = parentGasTarget - parentGasUsed;
+      const adjustedFeeDelta =
+        (baseFeePerGas * gasUsedDelta) /
+        parentGasTarget /
+        BlockParams.BASE_FEE_MAX_CHANGE_DENOMINATOR;
+      nextBaseFee = baseFeePerGas - adjustedFeeDelta;
+    }
+
+    return nextBaseFee;
+  }
+
+  static calcMaxNBlocksBaseFee(blocks: number, parentHeader: BaseFeeHeader) {
```

Comment:
I think the name `calcNBlocksMaxBaseFee` or `calcMaxBaseFeeNBlocks` would be better names for this function. The current name doesn't really convey what it returns, the maximum possible base fee after N blocks.